pkg/comp-functions/functions/vshnredis/script/prepRestore.sh

```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+
+set -euo pipefail
+
+source_namespace=$(kubectl -n "${CLAIM_NAMESPACE}" get vshnredis "${CLAIM_NAME}" -ojson | jq -r '.status.instanceNamespace')
+
+echo "copy secret"
+
+access_key=$(kubectl -n "${source_namespace}" get secret backup-bucket-credentials -o template='{{ .data.AWS_ACCESS_KEY_ID | base64decode }}')
+secret_key=$(kubectl -n "${source_namespace}" get secret backup-bucket-credentials -o template='{{ .data.AWS_SECRET_ACCESS_KEY | base64decode }}')
+restic_password=$(kubectl -n "${source_namespace}" get secret k8up-repository-password -o template='{{ .data.password | base64decode }}')
+restic_repository=$(kubectl -n "${source_namespace}" get snapshots.k8up.io "${BACKUP_NAME}" -o jsonpath='{.spec.repository}')
+backup_path=$(kubectl -n "${source_namespace}" get snapshots.k8up.io "${BACKUP_NAME}" -o jsonpath='{.spec.paths[0]}')
```

Comment:
What happens if the backup failed and the customer tries to use the backup to restore the instance?

Reply:
They probably either end up with a corrupted or an empty instance after the restore.

Reply:
I would try it in a manual test and see if we handle this better than having something that we don't know what might happen.

Reply:
Does a failed backup actually create a usable snapshot? If yes, shouldn't this be handled by k8up?

Reply:
As far as I can tell k8up snapshots have no info whether they failed, so I'd agree and expect that all k8up snapshots are valid and can be restored.

Reply:
There are two ways a backup could go wrong:

- the script as a whole fails and exits with != 0 -> then k8up notices that. however the underlying tool `restic` still creates a snapshot, as it doesn't know anything about the failure, it just takes everything it gets from stdin. It should be possible to teach K8up to remove the damaged snapshot.
- the script delivers k8up with garbage data and exits with 0 -> then k8up has no way to check if the data it received is actually usable

The first case should be easy to fix.

However the second case would probably be very hard to verify without actual restore tests.